src/main/jbake/content/blog/2016/10/14/hawkular-apm-polyglot-zipkin.adoc

```diff
@@ -0,0 +1,190 @@
+Zipkin http://zipkin.io/pages/existing_instrumentations.html[instrumentation libraries]
+and instrument polyglot applications. 
+
+In this blog post we will look how to use Zipkin instrumentation libraries to instrument 
+polyglot environment and how the reported data looks in Hawkular APM and Zipkin server. 
+
+== Polyglot Application
+A typical polyglot application consist of multiple applications written in different programing 
+languages. These applications interact together to fulfill some business needs or tasks. 
+
+The following image shows the architecture of our polyglot application.
+
+image::https://raw.githubusercontent.com/hawkular/hawkular-apm/master/examples/polyglot-zipkin/architecture.png[title="Architecture",align="center"]
+
+There are five applications written in 4 languages: Java, JavaScript, Python and Ruby. 
+It uses two databases MySQL and Apache Cassandra. Applications expose various REST endpoints 
+which simulates some behaviour. For example creating a user in a database or an asynchronous task.
+
+Each application lives in a separate docker container. The whole environment can be deployed 
+using one line docker-compose command. Overall there are seven containers including Apache Kafka 
+for collecting span data. The complete source code with instructions how to run it can be found on
+https://github.com/hawkular/hawkular-apm/tree/master/examples/polyglot-zipkin[Github].
+
+== Instrumentation Libraries
+There are several http://zipkin.io/pages/existing_instrumentations.html[instrumentation libraries]. 
+Developers have to choice one for targeted 
+language and framework they are using. If there is no instrumentation for a specific
+framework they have to report spans to the server manually, thus having an instrumentation
+library for a popular web framework is a big advantage (not limited to web).
+ 
+In a typical microservice environment it is also needed to add instrumentation for outgoing 
+requests so one application uses two instrumentations. One for incoming requests and one 
+for outgoing. These two instrumentation have to share context in order to accurately set id of 
+a parent span.
+
+Our polyglot applications are instrumented with following instrumentations:
+
+ * Java (JAX-RS): https://github.com/openzipkin/brave[brave]
+ * Java Dropwizard (JAX-RS): https://github.com/smoketurner/dropwizard-zipkin[dropwizard-zipkin] uses brave
+ * Apache Cassandra: https://github.com/thelastpickle/cassandra-zipkin-tracing[cassandra-zipkin-tracing] uses brave
+ * JavaScript (express, cujoJS): https://github.com/openzipkin/zipkin-js[zipkin-js]
+ * Ruby (rack): https://github.com/openzipkin/zipkin-ruby[zipkin-ruby]
+ * Python (Pyramid): https://github.com/Yelp/pyramid_zipkin[pyramid_zipkin]
+ * Python (Bravado): https://github.com/Yelp/swagger_zipkin[swagger_zipkin]
+
+These libraries are typically developed by a community and therefore reported data may differ. For 
```

Comment:
Added sentence: `It is not an invalid behaviour, just keep in mind when querying data or exporting to external systems for 
an analysis.`